### PR TITLE
add support for 'custom', per-customer data to be sent to LeadPerfection

### PR DESCRIPTION
**Pull request by cgrayson**

### CODE REVIEW

File: src/outbound.coffee [hunk 1]
```diff
@@ -39,6 +39,10 @@ request = (vars) ->
   body.Media6 = vars.leadperfection.media6 if vars.leadperfection?.media6?
   body.Media7 = vars.leadperfection.media7 if vars.leadperfection?.media7?
 
+  # add any "custom" fields that have been mapped to the root namespace
+  for name, value of vars
+    body[name] = value unless typeof value == 'object'
```
Comment (alexkwolfe) [inline src/outbound.coffee]: This could exclude a "rich data type" that has been mapped to these fields. Those are all objects.

**Comment by cgrayson** [comment]
I'm not thrilled with that, but it works. I suppose if `lead.` or an appended key included `raw` then it would include those too. :disappointed: Too bad the leadconduit-types don't all inherit from some parent class or share some other common attribute.

**Comment by cgrayson** [comment]
Or, @alexkwolfe, this problem goes away by forcing a namespace, like `custom.` :trollface:

**Comment by alexkwolfe** [comment]
Did you rule out something like below?

```coffeescript
for name, value of flat.flatten(vars, safe: true)
  body[name] = value unless value?.test(/^lead(perfection)?\./)
```

**Comment by alexkwolfe** [comment]
As we discussed today, asking the user to use `custom.AgentID` is a lot less intuitive than asking them to use `AgentID`. :hammer: :clock1030:

**Comment by alexkwolfe** [comment]
I was thinking about this some more. I'm not sure if this will ever be a requirement, but the code as it is written now doesn't support multiple values (i.e. `{ AgentID: ['1', '2'] }`) because arrays are excluded.

**Comment by cgrayson** [comment]
Implemented with test for a dot, that seems pretty clean. And it would work with an Array (or at least, it won't skip the array altogether).

**Comment by alexkwolfe** [comment]
LGTM

**Pull request merged**